### PR TITLE
Create certificates with not before/after years greater or equal 2050

### PR DESCRIPTION
**Pull request by sruetzler**
Should fix issue:
x509 certificate as PKCS #12 never valid if notAfter is in the year 2050 or later #530

### CODE REVIEW

File: tests/unit/x509.js [hunk 1]
```diff
@@ -406,6 +471,49 @@ var UTIL = require('../../lib/util');
...
+        name: 'localityName',
+        value: 'Blacksburg'
+      }, {
+        name: 'organizationName',
+        value: 'Test'
+      }, {
+        shortName: 'OU',
+        value: 'Test'
+      }];
+      let notBefore = new Date("2050-02-02");
```
Comment (davidlehn) [inline tests/unit/x509.js]: Use 'var' to fix the test suite.  We'll eventually move everything to ES6+ in one huge change.

File: tests/unit/x509.js [hunk 2]
```diff
@@ -406,6 +471,49 @@ var UTIL = require('../../lib/util');
...
+        subject: attrs,
+        issuer: attrs,
+        isCA: true,
+        notBefore : notBefore
+      });
+
+      var pem = PKI.certificateToPem(cert);
+      cert = PKI.certificateFromPem(pem);
+
+      let notAfter = new Date("2051-02-02");
```
Comment (davidlehn) [inline tests/unit/x509.js]: Use 'var'.

File: lib/x509.js [hunk 3]
```diff
@@ -2543,6 +2541,21 @@ pki.getTBSCertificate = function(cert) {
   return tbs;
 };
 
+const jan_1_1950 = new Date('1950-01-01T00:00:00Z');
+const jan_1_2050 = new Date('2050-01-01T00:00:00Z');
+
+function dateToPem(date){
```
Comment (davidlehn) [inline lib/x509.js]: This is going to ASN1.  Should rename it to `_dateToAsn1` and perhaps move it near one of the other `_*ToAsn1` helper functions.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "node-forge",
-  "version": "0.7.7-dev",
+  "version": "0.7.6-1",
```
Comment (davidlehn) [inline package.json]: Would rather not have this in the patch since we try to keep the -dev suffix for unreleased code.

**Comment by sruetzler** [comment]
I close the pull request and open another with the requested changes

**Pull request closed without merging**